urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/driver/Phase1.java

```diff
@@ -0,0 +1,244 @@
+        String format = "%-9s --> %9s : %s%n";
+        System.out.format(format, "$10.50", testPriceHolder.get(0), testPriceHolder.get(0).toString().equals("$10.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$1,400.99", testPriceHolder.get(1), testPriceHolder.get(1).toString().equals("$1,400.99") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-51.52", testPriceHolder.get(2), testPriceHolder.get(2).toString().equals("$-51.52") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$0.49", testPriceHolder.get(3), testPriceHolder.get(3).toString().equals("$0.49") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$-0.89", testPriceHolder.get(4), testPriceHolder.get(4).toString().equals("$-0.89") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$12.00", testPriceHolder.get(5), testPriceHolder.get(5).toString().equals("$12.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$90.00", testPriceHolder.get(6), testPriceHolder.get(6).toString().equals("$90.00") ? "CORRECT" : "ERROR");
+        System.out.format(format, "$14.50", testPriceHolder.get(7), testPriceHolder.get(7).toString().equals("$14.50") ? "CORRECT" : "ERROR");
+        System.out.format(format, "MKT", testPriceHolder.get(8), testPriceHolder.get(8).toString().equals("MKT") ? "CORRECT" : "ERROR");
+        System.out.println();
+    }
+
+    private static void verifyMathematicalOperations() {
+        System.out.println("Verifying the Functionality of your Mathematical Operations:");
+        String format = "%-9s %c %9s = %9s : %s%n";
+        try {
+            Price results = testPriceHolder.get(0).add(testPriceHolder.get(1));
+            System.out.format(format, testPriceHolder.get(0), '+', testPriceHolder.get(1), results, results.toString().equals("$1,411.49") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(1).subtract(testPriceHolder.get(1));
+            System.out.format(format, testPriceHolder.get(1), '-', testPriceHolder.get(1), results, results.toString().equals("$0.00") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(2).add(testPriceHolder.get(3));
+            System.out.format(format, testPriceHolder.get(2), '+', testPriceHolder.get(3), results, results.toString().equals("$-51.03") ? "CORRECT" : "ERROR");
+        } catch (InvalidPriceOperation ex) {
+            System.out.println("FAILED: " + ex.getMessage());
+        }
+        try {
+            Price results = testPriceHolder.get(3).multiply(4);
+            System.out.format(format, testPriceHolder.get(3), '*', 4, results, results.toString().equals("$1.96") ? "CORRECT" : "ERROR");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: . This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 79. Consider formatting this data with &#39;%d&#39; instead.